Bottom: dfdeec2f39d5f208588fe8939fa8ae4c6025fa09
Top:    cf831a3507ae70ed677cd37d9a20887710b4d16a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-24 21:13:12 +0100

Use motion hints when processing pointer motion over the drawing area


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 68f39c8..6b65498 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -697,7 +697,7 @@ ghid_show_crosshair (gboolean show)
 
   glBegin (GL_LINES);
 
-#if 1
+#if 0
   if (x_prev >= 0)
     draw_crosshair (x_prev, y_prev);
 #endif
@@ -710,7 +710,7 @@ ghid_show_crosshair (gboolean show)
 
   glBegin (GL_QUADS);
 
-#if 1
+#if 0
   if (x_prev >= 0 && draw_markers_prev)
     {
       glVertex2i (0,                  y_prev - VCD);
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7cd464a..0b1f72a 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -682,12 +682,14 @@ queue_tooltip_update (GHidPort *out)
 
 gint
 ghid_port_window_motion_cb (GtkWidget * widget,
-			    GdkEventButton * ev, GHidPort * out)
+			    GdkEventMotion * ev, GHidPort * out)
 {
   gdouble dx, dy;
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
 
+  gdk_event_request_motions (ev);
+
 #if 0
   if (!ghid_start_drawing (out))
     return FALSE;
@@ -696,7 +698,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   if (out->panning)
     {
 /* Not sure this really matters, since we're using invalidate to get a redraw */
-#if 1
+#if 0
       if (gtk_events_pending ())
 	return FALSE;
 #endif
@@ -709,7 +711,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
       return FALSE;
     }
   x_prev = y_prev = -1;
-  moved = ghid_note_event_location (ev);
+  moved = ghid_note_event_location ((GdkEventButton *)ev);
 
 #if ENABLE_TOOLTIPS
   queue_tooltip_update (out);
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 7c5823f..168fd1a 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -2300,7 +2300,8 @@ ghid_build_pcb_top_window (void)
 			 | GDK_LEAVE_NOTIFY_MASK | GDK_ENTER_NOTIFY_MASK
 			 | GDK_BUTTON_RELEASE_MASK | GDK_BUTTON_PRESS_MASK
 			 | GDK_KEY_RELEASE_MASK | GDK_KEY_PRESS_MASK
-			 | GDK_FOCUS_CHANGE_MASK | GDK_POINTER_MOTION_MASK);
+			 | GDK_FOCUS_CHANGE_MASK | GDK_POINTER_MOTION_MASK
+			 | GDK_POINTER_MOTION_HINT_MASK);
 
   /*
    * This is required to get the drawing_area key-press-event.  Also the
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 7ea38c5..2eba31c 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -305,7 +305,7 @@ gint ghid_port_window_enter_cb (GtkWidget * widget,
 gint ghid_port_window_leave_cb (GtkWidget * widget, 
                            GdkEventCrossing * ev, GHidPort * out);
 gint ghid_port_window_motion_cb (GtkWidget * widget,
-				 GdkEventButton * ev, GHidPort * out);
+				 GdkEventMotion * ev, GHidPort * out);
 gint ghid_port_window_mouse_scroll_cb (GtkWidget * widget,
 				       GdkEventScroll * ev, GHidPort * out);
